<launch>
  <arg name="manager" default="/camera_nodelet_manager" />

  <include file="$(find openni_launch)/launch/openni.launch">
    <param name="depth_registered" value="true"/>
  </include>

  <arg name="input_point" default="/camera/depth_registered/points" />
  <group ns="pcl_nodelet">
    <node pkg="nodelet" type="nodelet"
          name="voxelgrid"
          args="load pcl/VoxelGrid $(arg manager)"
          output="screen" clear_params="true">
      <remap from="~input" to="$(arg input_point)" />
      <rosparam>
        filter_limit_max: 5.0
        leaf_size: 0.01
      </rosparam>
    </node>
    <node pkg="nodelet" type="nodelet"
          name="sac_segmentation"
          args="load pcl/SACSegmentation $(arg manager)"
          output="screen" clear_params="true">
      <remap from="~input" to="/pcl_nodelet/voxelgrid/output" />
      <rosparam>
        model_type: 0
        distance_threshold: 0.01
      </rosparam>
    </node>
    <node pkg="nodelet" type="nodelet"
          name="extract_indices"
          args="load pcl/ExtractIndices $(arg manager)"
          output="screen" clear_params="true">
      <remap from="~input" to="/pcl_nodelet/voxelgrid/output" />
      <remap from="~indices" to="/pcl_nodelet/sac_segmentation/inliers" />
      <rosparam>
        model_type: 0
        distance_threshold: 0.01
        negative: true
      </rosparam>
    </node>
    <node pkg="nodelet" type="nodelet"
          name="centroid_publisher"
          args="load jsk_pcl/CentroidPublisher $(arg manager)"
          output="screen" clear_params="true">
      <remap from="~input" to="/pcl_nodelet/extract_indices/output" />
      <rosparam>
        frame: result_center
      </rosparam>
    </node>
  </group>
</launch>